LLMonFHIR/FHIRSummary/FHIRResourceSummary.swift

```diff
@@ -21,31 +21,44 @@ final class FHIRResourceSummary: Sendable {
         let title: String
         /// Summary of the FHIR resource, should be a single line of text.
         let summary: String
-        
-        
+        /// Indicates if this summary used fallback values due to parsing issues.
+        let needsRetry: Bool
+
         var description: String {
             title + "\n" + summary
         }
+
         
-        
-        init?(_ description: String) {
-            let components = description.split(separator: "\n")
-            guard components.count == 2, let title = components.first, let summary = components.last else {
-                return nil
+        init(_ description: String) {
+            let lines = description.components(separatedBy: "\n")
+            let nonEmptyLines = lines.filter { !$0.isEmpty }
+
+            switch nonEmptyLines.count {
+            case 0:
+                title = "Unknown Resource"
+                summary = "Failed to generate summary"
+                needsRetry = true
```

Comment:
This is slightly different from what we had. Let me know what do you think? Is this case is hit, this will auto-retry one time.